File: Secure Code Assessment/assets/vulnerable_sample.js
const express = require('express');
const mysql = require('mysql');
const child_process = require('child_process');
const crypto = require('crypto');

const app = express();

// HARDCODED API KEYS AND SECRETS
const API_KEY = "sk_live_994821904128409128490";
const DATABASE_URI = "mongodb://admin:P@ssword2026@localhost:27017/prod_db";

const db = mysql.createConnection({
    host: 'localhost',
    user: 'root',
    password: 'root_password_123',
    database: 'secure_app'
});

// SQL INJECTION
app.get('/api/users', (req, res) => {
    const userId = req.query.id;
    // Vulnerable SQL concatenation
    const query = "SELECT * FROM users WHERE id = " + userId;
    db.query(query, (err, results) => {
        if (err) throw err;
        res.json(results);
    });
});

// CROSS-SITE SCRIPTING (XSS) & DANGEROUS EVAL
app.post('/api/evaluate', (req, res) => {
    const expression = req.body.expression;
    // Use of dangerous eval function
    const result = eval(expression);
    
    // Direct innerHTML style response injection
    res.send(`<div id="result">Result: ${req.body.username} is ${result}</div>`);
});

// COMMAND INJECTION
app.get('/api/exec', (req, res) => {
    const userCmd = req.query.cmd;
    // Executing user controlled input directly
    child_process.exec('ls -la ' + userCmd, (error, stdout, stderr) => {
        res.send(stdout);
    });
});

// WEAK CRYPTOGRAPHY (MD5 & SHA1)
function hashPassword(password) {
    // Deprecated weak hash algorithm for security critical password storage
    return crypto.createHash('md5').update(password).digest('hex');
}

// INSECURE SENSITIVE INFORMATION LOGGING
app.post('/login', (req, res) => {
    console.log("Attempted login with password:", req.body.password);
    console.log("Session auth token:", req.headers['authorization']);
});

app.listen(3000, () => {
    console.log("Server listening on port 3000");
});

File: Secure Code Assessment/assets/vulnerable_sample.cpp
#include <iostream>
#include <cstdlib>
#include <cstring>
#include <cstdio>

// HARDCODED SECRETS
#define DATABASE_PASSWORD "HardcodedCppPass99#"
#define SYSTEM_ENCRYPTION_KEY "0x4F9A2B81E73C"

void processUserData(const char* input) {
    char buffer[64];
    
    // BUFFER OVERFLOW / UNSAFE MEMORY COPIES
    strcpy(buffer, input); // Dangerous strcpy without length validation
    
    // COMMAND INJECTION
    char command[128];
    sprintf(command, "echo 'User input logged: %s' >> log.txt", input);
    system(command); // Unchecked system command invocation
}

void queryDatabase(const char* username) {
    char sqlQuery[256];
    // SQL INJECTION PATTERN
    sprintf(sqlQuery, "SELECT * FROM accounts WHERE user = '%s';", username);
    std::cout << "Executing Query: " << sqlQuery << std::endl;
}

int main(int argc, char* argv[]) {
    if (argc > 1) {
        processUserData(argv[1]);
        queryDatabase(argv[1]);
    }
    
    // WEAK RANDOM NUMBER GENERATOR FOR SECURITY TOKENS
    int token = rand() % 10000;
    std::cout << "Session Security Token: " << token << std::endl;
    
    return 0;
}
